src/task_queue/database_tasks/translate_db_content.py

```diff
@@ -0,0 +1,92 @@
+from django.apps import apps
+from django.forms import model_to_dict
+from django.utils import timezone
+
+from _main_.utils.massenergize_logger import log
+from _main_.utils.translation import JsonTranslator
+from database.models import SupportedLanguage, TranslationsCache
+
+SOURCE_LANGUAGE_CODE = 'en'
+
+
+class TranslationError(Exception):
+	"""Raised when an error occurs during translation."""
+
+
+class TranslateDBContents:
+	"""
+	Class responsible for translating database and static json contents.
+	It uses a caching mechanism to avoid translating the same text multiple times.
+	"""
+	
+	def __init__(self):
+		self.translator = JsonTranslator
+		self.supported_languages = SupportedLanguage.objects.values_list('code', flat=True)  #TODO: Use George's unitils to get supported codes
+	
+	def cache_translations(self, hashes, translated_text_list, language) -> bool:
+		try:
+			caches = []
+			for _hash, translated_text in zip(hashes, translated_text_list):
+				cache = TranslationsCache(
+					hash=_hash,
+					target_language_code=language,
+					source_language_code=SOURCE_LANGUAGE_CODE,
+					translated_text=translated_text
+				)
+				caches.append(cache)
+			TranslationsCache.objects.bulk_create(caches)
+			return True
+		except Exception as e:
+			log.exception(e, extra={'MESSAGE_ID': 'TRANSLATION_ERROR'})
+			return False
+	
+	def get_valid_instances(self, model) -> list:
+		query_params = {}
+		if hasattr(model, 'is_deleted'):
+			query_params['is_deleted'] = False
+		if hasattr(model, 'is_archived'):
+			query_params['is_archived'] = False
+		if hasattr(model, 'is_published'):
+			query_params['is_published'] = True
+		if hasattr(model, 'is_active'):
+			query_params['is_active'] = True
+		return model.objects.filter(**query_params)
+	
+	def load_db_contents_and_translate(self) -> bool:
```

Comment:
i think this function is a repetition of something that George has.  Maybe sync on this?  check https://github.com/massenergize/api/pull/1122/files `create_list_of_all_records_to_translate `

Reply:
George and I decided to move the function to the utils.py file in _main_ so that each of us can use it in our code.